rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -233,11 +234,17 @@ void TimeSource::create_clock_sub()
     return;
   }
 
+  using rclcpp::QosPolicyKind;
+  rclcpp::SubscriptionOptions options;
+  options.qos_overriding_options = QosOverridingOptions{
+    QosPolicyKind::Depth, QosPolicyKind::History, QosPolicyKind::LivelinessLeaseDuration,
+    QosPolicyKind::Reliability};
```

Comment:
The only reason I didn't add durability here is that the publisher in `gazebo_ros_packages` isn't offering transient local durability and that isn't currently reconfigurable.

> If we are going ahead an setting options for the /clock topic here, I think we should add rationale to the design doc.

I think that what QoS should be reconfigurable for each topic can be discussed in a case by case fashion, without adding that to the design doc.

Reply:
It might be nice to document QoS options for common topics so that they can match in multiple client libraries; i.e. I presume we want rclcpp, rclpy, etc to have the same settings.

Reply:
> I presume we want rclcpp, rclpy, etc to have the same settings.

That's a fair point.

Reply:
I have deleted that change, so that can be discussed in a separated PR: https://github.com/ros2/rclcpp/pull/1408/commits/7bb8462b3c93967413163c5dd7aef1a3b12f0aa9.